### PR TITLE
Set prevent_destroy to true for nodes

### PR DESCRIPTION
**Pull request by rxbn**
This PR makes the Terraform lifecycle option "prevent_destroy" configurable and sets it to `true` by default.
This is done so that you have to consciously disable it for certain node groups.

## Checklist

- [x] Keep pull requests small so they can be easily reviewed.
- [x] Categorize the PR by setting a good title and adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog
- [x] Link this PR to related issues.

### CODE REVIEW

**Review by megian — APPROVED** [review]

File: modules/node-group/providers.tf [hunk 1]
```diff
@@ -1,5 +1,5 @@
 terraform {
-  required_version = ">= 0.14"
+  required_version = ">= 1.3.0"
```
Comment (megian) [inline modules/node-group/providers.tf:2]: Is this relevant for the prevent_destroy? If not, I would higly suggest to do this in an own PR, to make it visible.

Comment (rxbn) [reply modules/node-group/providers.tf:2]: No, we can make it in a separate PR. Will remove it from here.

**Review by megian — APPROVED** [review]
